qiskit/qasm3/ast.py

```diff
@@ -173,6 +173,18 @@ class BitType(ClassicalType):
     __slots__ = ()
 
 
+class DurationType(ClassicalType):
+    """Type information for a duration."""
+
+    __slots__ = ()
+
+
+class StretchType(ClassicalType):
+    """Type information for a stretch."""
+
+    __slots__ = ()
+
+
```

Comment:
Maybe just semantics: my reading was we give-or-take decided to interpret the OQ3 spec as saying that `stretch x;` didn't involve a _type_ `stretch` so much as just being a special keyword for a `duration`-typed declaration.  I'm also fine to leave it the way we have (heaven knows this AST is a jumble), but in another world: would it make more sense to handle `stretch x (= Expr)?;` as a
```python
class StretchDeclaration(Statement):
    identifier: Identifier
    bound: Optional[Expr]
```
rather than a type?

Reply:
Ah yeah, that makes a ton more sense. We don't use this in this PR anyway, so if it's alright, I'll leave it for now and make the switch in the PR that actually does.